### DISCUSSION TITLE
Runtime configuration of kinematics plugins via ROS parameters

### DISCUSSION THREAD

**Discussion by galou** [moveit/moveit2#2888] [General]
I have a kinematics plugin that has some ROS parameters. When I create an instance of the plugin directly in my control node, I can change the parameters during runtime. I didn't manage to change the parameter of the instance created by `moveit_simple_controller_manager`. I can read it with
```shell
$ ros2 param get /moveit_simple_controller_manager with_experimental_joint_limits 
Boolean value is: False
```

but when I attempt to change it, I get
```
$ ros2 param set /moveit_simple_controller_manager with_experimental_joint_limits false
Setting parameter failed
```

Is this a feature? How can I change the behavior of my kinematics plugin in runtime then?

**Comment by galou** [comment]
Anyone to help on this, please?

**Comment by mikeferguson** [comment]
Looking at the code for MoveItSimpleControllerManager it does pretty much everything parameter related in the initalize() function, and does not set a parameter change callback - so yeah, I wouldn't expect this to work for setting any of the configuration parameters.

As for your plugin, I have no idea what that code has going on inside - but if it doesn't call add_on_set_parameters_callback to setup a callback that handles the parameter change, it would also not work.

**Comment by galou** [comment]
I use [generate_parameter_library](https://github.com/PickNikRobotics/generate_parameter_library), so the callbacks should all be set. As I wrote, everything works as expected when I create an instance of the plugin directly in my node.
